File: app/src/main/java/com/giaquino/android/cropped/CroppedApplication.java
package com.giaquino.android.cropped;

import android.app.Application;

import com.giaquino.android.cropped.di.AppContainer;

import timber.log.Timber;

public class CroppedApplication extends Application {

    public AppContainer appContainer;

    @Override
    public void onCreate() {
        super.onCreate();
        Timber.plant(new Timber.DebugTree());
        initializeContainers();
    }

    private void initializeContainers() {
        appContainer = new AppContainer();
    }
}

File: app/src/main/java/com/giaquino/android/cropped/CroppedApplication.kt
package com.giaquino.android.cropped

import dagger.hilt.android.HiltAndroidApp
import android.app.Application
import android.content.Context
import androidx.datastore.core.DataStore
import androidx.datastore.preferences.core.Preferences
import androidx.datastore.preferences.preferencesDataStore
import timber.log.Timber
import timber.log.Timber.DebugTree

val Context.accountDataStore : DataStore<Preferences> by preferencesDataStore(name = "account.ds")

@HiltAndroidApp
class CroppedApplication : Application() {

    override fun onCreate() {
        super.onCreate()
        Timber.plant(DebugTree())
    }
}